Unsupported Qdrant value type when using Qdrant with VectorStoreChatMemoryAdvisor

Starting with RC1, I get the following when using OpenAI, `VectorStoreChatMemoryAdvisor` and a Qdrant vector store:

```
java.lang.RuntimeException: java.lang.IllegalArgumentException: Unsupported Qdrant value type: class java.util.ArrayList
	at org.springframework.ai.vectorstore.qdrant.QdrantVectorStore.toPayload(QdrantVectorStore.java:304) ~[spring-ai-qdrant-store-1.0.0-RC1.jar:1.0.0-RC1]
	at org.springframework.ai.vectorstore.qdrant.QdrantVectorStore.lambda$doAdd$0(QdrantVectorStore.java:186) ~[spring-ai-qdrant-store-1.0.0-RC1.jar:1.0.0-RC1]
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:212) ~[na:na]
	at java.base/java.util.AbstractList$RandomAccessSpliterator.forEachRemaining(AbstractList.java:722) ~[na:na]
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:556) ~[na:na]
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:546) ~[na:na]
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:622) ~[na:na]
	at java.base/java.util.stream.AbstractPipeline.evaluateToArrayNode(AbstractPipeline.java:291) ~[na:na]
	at java.base/java.util.stream.ReferencePipeline.toArray(ReferencePipeline.java:631) ~[na:na]
	at java.base/java.util.stream.ReferencePipeline.toArray(ReferencePipeline.java:637) ~[na:na]
	at java.base/java.util.stream.ReferencePipeline.toList(ReferencePipeline.java:642) ~[na:na]
	at org.springframework.ai.vectorstore.qdrant.QdrantVectorStore.doAdd(QdrantVectorStore.java:188) ~[spring-ai-qdrant-store-1.0.0-RC1.jar:1.0.0-RC1]
	at org.springframework.ai.vectorstore.observation.AbstractObservationVectorStore.lambda$add$1(AbstractObservationVectorStore.java:85) ~[spring-ai-vector-store-1.0.0-RC1.jar:1.0.0-RC1]
	at io.micrometer.observation.Observation.observe(Observation.java:498) ~[micrometer-observation-1.14.6.jar:1.14.6]
	at org.springframework.ai.vectorstore.observation.AbstractObservationVectorStore.add(AbstractObservationVectorStore.java:85) ~[spring-ai-vector-store-1.0.0-RC1.jar:1.0.0-RC1]
	at org.springframework.ai.vectorstore.VectorStore.accept(VectorStore.java:56) ~[spring-ai-vector-store-1.0.0-RC1.jar:1.0.0-RC1]
	at org.springframework.ai.vectorstore.VectorStore.accept(VectorStore.java:41) ~[spring-ai-vector-store-1.0.0-RC1.jar:1.0.0-RC1]
	at org.springframework.ai.document.DocumentWriter.write(DocumentWriter.java:30) ~[spring-ai-commons-1.0.0-RC1.jar:1.0.0-RC1]
	at org.springframework.ai.chat.client.advisor.vectorstore.VectorStoreChatMemoryAdvisor.after(VectorStoreChatMemoryAdvisor.java:165) ~[spring-ai-advisors-vector-store-1.0.0-RC1.jar:1.0.0-RC1]
	at org.springframework.ai.chat.client.advisor.api.BaseAdvisor.adviseCall(BaseAdvisor.java:53) ~[spring-ai-client-chat-1.0.0-RC1.jar:1.0.0-RC1]
	at org.springframework.ai.chat.client.advisor.DefaultAroundAdvisorChain.lambda$nextCall$1(DefaultAroundAdvisorChain.java:110) ~[spring-ai-client-chat-1.0.0-RC1.jar:1.0.0-RC1]
	at io.micrometer.observation.Observation.observe(Observation.java:564) ~[micrometer-observation-1.14.6.jar:1.14.6]
	at org.springframework.ai.chat.client.advisor.DefaultAroundAdvisorChain.nextCall(DefaultAroundAdvisorChain.java:110) ~[spring-ai-client-chat-1.0.0-RC1.jar:1.0.0-RC1]
	at org.springframework.ai.chat.client.DefaultChatClient$DefaultCallResponseSpec.lambda$doGetObservableChatClientResponse$1(DefaultChatClient.java:467) ~[spring-ai-client-chat-1.0.0-RC1.jar:1.0.0-RC1]
	at io.micrometer.observation.Observation.observe(Observation.java:564) ~[micrometer-observation-1.14.6.jar:1.14.6]
	at org.springframework.ai.chat.client.DefaultChatClient$DefaultCallResponseSpec.doGetObservableChatClientResponse(DefaultChatClient.java:465) ~[spring-ai-client-chat-1.0.0-RC1.jar:1.0.0-RC1]
	at org.springframework.ai.chat.client.DefaultChatClient$DefaultCallResponseSpec.doGetObservableChatClientResponse(DefaultChatClient.java:446) ~[spring-ai-client-chat-1.0.0-RC1.jar:1.0.0-RC1]
	at org.springframework.ai.chat.client.DefaultChatClient$DefaultCallResponseSpec.content(DefaultChatClient.java:441) ~[spring-ai-client-chat-1.0.0-RC1.jar:1.0.0-RC1]
	at com.broadcom.tanzu.ragqdrant.RagQdrantApplication.lambda$go$0(RagQdrantApplication.java:58) ~[classes/:na]
	at org.springframework.boot.SpringApplication.lambda$callRunner$4(SpringApplication.java:785) ~[spring-boot-3.4.5.jar:3.4.5]
	at org.springframework.util.function.ThrowingConsumer$1.acceptWithException(ThrowingConsumer.java:82) ~[spring-core-6.2.6.jar:6.2.6]
	at org.springframework.util.function.ThrowingConsumer.accept(ThrowingConsumer.java:60) ~[spring-core-6.2.6.jar:6.2.6]
	at org.springframework.util.function.ThrowingConsumer$1.accept(ThrowingConsumer.java:86) ~[spring-core-6.2.6.jar:6.2.6]
	at org.springframework.boot.SpringApplication.callRunner(SpringApplication.java:797) ~[spring-boot-3.4.5.jar:3.4.5]
	at org.springframework.boot.SpringApplication.callRunner(SpringApplication.java:785) ~[spring-boot-3.4.5.jar:3.4.5]
	at org.springframework.boot.SpringApplication.lambda$callRunners$3(SpringApplication.java:773) ~[spring-boot-3.4.5.jar:3.4.5]
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:184) ~[na:na]
	at java.base/java.util.stream.SortedOps$SizedRefSortingSink.end(SortedOps.java:357) ~[na:na]
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:557) ~[na:na]
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:546) ~[na:na]
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:151) ~[na:na]
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:174) ~[na:na]
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:265) ~[na:na]
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:611) ~[na:na]
	at org.springframework.boot.SpringApplication.callRunners(SpringApplication.java:773) ~[spring-boot-3.4.5.jar:3.4.5]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:325) ~[spring-boot-3.4.5.jar:3.4.5]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1362) ~[spring-boot-3.4.5.jar:3.4.5]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1351) ~[spring-boot-3.4.5.jar:3.4.5]
	at com.broadcom.tanzu.ragqdrant.RagQdrantApplication.main(RagQdrantApplication.java:23) ~[classes/:na]
Caused by: java.lang.IllegalArgumentException: Unsupported Qdrant value type: class java.util.ArrayList
	at org.springframework.ai.vectorstore.qdrant.QdrantValueFactory.value(QdrantValueFactory.java:80) ~[spring-ai-qdrant-store-1.0.0-RC1.jar:1.0.0-RC1]
	at org.springframework.ai.vectorstore.qdrant.QdrantValueFactory.lambda$toValueMap$1(QdrantValueFactory.java:45) ~[spring-ai-qdrant-store-1.0.0-RC1.jar:1.0.0-RC1]
	at java.base/java.util.stream.Collectors.lambda$uniqKeysMapAccumulator$1(Collectors.java:180) ~[na:na]
	at java.base/java.util.stream.ReduceOps$3ReducingSink.accept(ReduceOps.java:169) ~[na:na]
	at java.base/java.util.HashMap$EntrySpliterator.forEachRemaining(HashMap.java:1858) ~[na:na]
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:556) ~[na:na]
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:546) ~[na:na]
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:921) ~[na:na]
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:265) ~[na:na]
	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:702) ~[na:na]
	at org.springframework.ai.vectorstore.qdrant.QdrantValueFactory.toValueMap(QdrantValueFactory.java:45) ~[spring-ai-qdrant-store-1.0.0-RC1.jar:1.0.0-RC1]
	at org.springframework.ai.vectorstore.qdrant.QdrantVectorStore.toPayload(QdrantVectorStore.java:299) ~[spring-ai-qdrant-store-1.0.0-RC1.jar:1.0.0-RC1]
	... 48 common frames omitted
```

This appears to have been introduced with https://github.com/spring-projects/spring-ai/commit/6c9b4d6cd49d173bd3b806c98c1afff2b4ff76e1. 

The following code snippet reproduces it:

```
var chatClient = chatClientBuilder
        .defaultAdvisors(
                VectorStoreChatMemoryAdvisor.builder(vectorStore).build())
        .build();
var answer = chatClient.prompt()
        .user("How many can play the game?")
        .call()
        .content();
System.err.println(answer);
```

--- a/vector-stores/spring-ai-qdrant-store/src/main/java/org/springframework/ai/vectorstore/qdrant/QdrantValueFactory.java
+++ b/vector-stores/spring-ai-qdrant-store/src/main/java/org/springframework/ai/vectorstore/qdrant/QdrantValueFactory.java
@@ -32,6 +32,7 @@ import org.springframework.util.Assert;
  * Utility methods for building io.qdrant.client.grpc.JsonWithInt.Value from Java objects.
  *
  * @author Anush Shetty
+ * @author Ilayaperumal Gopinathan
  * @since 0.8.1
  */
 final class QdrantValueFactory {
@@ -65,6 +66,10 @@ final class QdrantValueFactory {
 			return value((Map<String, Object>) value);
 		}
 
+		if (value instanceof List) {
+			return value((List<Object>) value);
+		}
+
 		switch (value.getClass().getSimpleName()) {
 			case "String":
 				return ValueFactory.value((String) value);
@@ -79,6 +84,16 @@ final class QdrantValueFactory {
 			default:
 				throw new IllegalArgumentException("Unsupported Qdrant value type: " + value.getClass());
 		}
+	}
+
+	private static Value value(List<Object> elements) {
+		List<Value> values = new ArrayList<Value>(elements.size());
+
+		for (Object element : elements) {
+			values.add(value(element));
+		}
+
+		return ValueFactory.list(values);
 	}
 
 	private static Value value(Object[] elements) {

--- a/vector-stores/spring-ai-qdrant-store/src/test/java/org/springframework/ai/vectorstore/qdrant/QdrantVectorStoreIT.java
+++ b/vector-stores/spring-ai-qdrant-store/src/test/java/org/springframework/ai/vectorstore/qdrant/QdrantVectorStoreIT.java
@@ -79,10 +79,11 @@ public class QdrantVectorStoreIT extends BaseVectorStoreTests {
 	List<Document> documents = List.of(
 			new Document("Spring AI rocks!! Spring AI rocks!! Spring AI rocks!! Spring AI rocks!! Spring AI rocks!!",
 					Collections.singletonMap("meta1", "meta1")),
-			new Document("Hello World Hello World Hello World Hello World Hello World Hello World Hello World"),
+			new Document("Hello World Hello World Hello World Hello World Hello World Hello World Hello World",
+					Collections.singletonMap("meta1", List.of("meta-list"))),
 			new Document(
 					"Great Depression Great Depression Great Depression Great Depression Great Depression Great Depression",
-					Collections.singletonMap("meta2", "meta2")));
+					Collections.singletonMap("meta2", List.of("meta-list"))));
 
 	@BeforeAll
 	static void setup() throws InterruptedException, ExecutionException {

